sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/BufferStagingArea.java

```diff
@@ -90,14 +90,38 @@ public Flux<BufferAggregator> write(ByteBuffer buf) {
 
             result = Flux.just(this.currentBuf);
 
+            // calculate how many chunks remain given the total size of the buffer and the staging area window
```

Comment:
We should probably add a comment to overall note that this logic is to handle when the buffer provided is larger than the remaining capacity + the fixed sized of a new BufferAggregator. We can even probably use some of the wording in explaining it from the comment that I mention we should remove later on how we don't handle this sort of overflow.